flask_pydantic_api/api_wrapper.py

```diff
@@ -212,7 +220,10 @@ def wrapped_endpoint(*args: Any, **kwargs: Any) -> Callable:
                             result.__class__, success_status_code
                         )
 
-                    result = make_response(result_data, status_code)
+                    response = make_response(result_data, status_code)
+
+                else:
+                    response = make_response(result)
```

Comment:
This is a functional change on top of the typing changes.  Please explain the rationale here.

Reply:
Basically it is quite difficult to completely specify the return type otherwise. I'll take another stab at it if I think the change has a chance of getting merged.